src/main/scala/chisel3/testers/TesterDriver.scala

```diff
@@ -74,15 +125,76 @@ object TesterDriver extends BackendCompilationUtilities {
       false
     }
   }
+
+  def executeTreadle(t:                    () => BasicTester,
+                     additionalVResources: Seq[String] = Seq(),
+                     annotations:          AnnotationSeq = Seq(),
+                     nameHint:             Option[String] = None): Boolean = {
+    val generatorAnnotation = chisel3.stage.ChiselGeneratorAnnotation(t)
+
+    // This provides an opportunity to translate from top level generic flags to backend specific annos
+    var annotationSeq = annotations :+ WriteVcdAnnotation
+
+    // This produces a chisel circuit annotation, a later pass will generate a firrtl circuit
+    // Can't do both at once currently because generating the latter deletes the former
+    annotationSeq = (new chisel3.stage.phases.Elaborate).transform(annotationSeq :+ generatorAnnotation)
+
+    val circuit = annotationSeq.collect { case x: ChiselCircuitAnnotation => x }.head.circuit
+
+//    val targetName = s"test_run_dir/${circuit.name}" + (nameHint match {
+//      case Some(hint) => s"_$hint"
+//      case _          => ""
+//    }) + "_treadle"
+
+    val targetName: File = createTestDirectory(circuit.name)
+
+    annotationSeq = annotationSeq :+ TargetDirAnnotation(targetName.getPath) // :+ ClockInfoAnnotation(Seq(ClockInfo("clock", 10, 0))) // :+ CallResetAtStartupAnnotation
+    annotationSeq = annotationSeq :+ TargetDirAnnotation(targetName.getPath) :+ CallResetAtStartupAnnotation
```

Comment:
is the duplication of TargetDirAnnotation intentional?

Reply:
Fixed